lldb/source/Plugins/Process/elf-core/ProcessElfCore.cpp

```diff
@@ -277,20 +277,36 @@ Status ProcessElfCore::DoLoadCore() {
 
 void ProcessElfCore::UpdateBuildIdForNTFileEntries() {
   Log *log = GetLog(LLDBLog::Process);
+  m_uuids.clear();
   for (NT_FILE_Entry &entry : m_nt_file_entries) {
     entry.uuid = FindBuidIdInCoreMemory(entry.start);
-    if (log && entry.uuid.IsValid())
-      LLDB_LOGF(log, "%s found UUID @ %16.16" PRIx64 ": %s \"%s\"",
-                __FUNCTION__, entry.start, entry.uuid.GetAsString().c_str(),
-                entry.path.c_str());
+    if (entry.uuid.IsValid()) {
+      m_uuids[entry.path] = entry.uuid;
+      if (log)
+        LLDB_LOGF(log, "%s found UUID @ %16.16" PRIx64 ": %s \"%s\"",
+                  __FUNCTION__, entry.start, entry.uuid.GetAsString().c_str(),
+                  entry.path.c_str());
+    }
   }
 }
 
+llvm::StringRef ProcessElfCore::GetMainExecutablePath() {
+  // The first entry in the NT_FILE might be our executable
+  llvm::StringRef executable_path = m_nt_file_entries[0].path;
```

Comment:
Right now we are calling this function from a context where we have checked `!m_nt_file_entries.empty()` so it is safe to index the 0th element in the `m_nt_files_entries` array, but that might not always be the case. We should make this function safe even if `m_nt_file_entries` is empty by returning an empty string.